cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala

```diff
@@ -164,11 +164,11 @@ class QueryPlanTest extends DocumentingTestBase with SoftReset {
     profileQuery(
       title = "Eager",
       text =
-        """For isolation purposes this operator makes sure that operations that affect subsequent operations are executed fully for the whole dataset before continuing execution.
-           | Otherwise it could trigger endless loops, matching data again, that was just created.
+        """For isolation purposes, the `Eager` operator ensures that operations affecting subsequent operations are executed fully for the whole dataset before continuing execution.
+           | Otherwise, endless loops could be triggered in which data that was just created is matched.
```

Comment:
> Otherwise, endless loops could be triggered in which data that was just created is matched.

Information from the stores is fetched in a lazy manner, i.e. the pattern matching might not be fully exhausted before we start applying updates. To keep the semantics sane, the query planner will insert Eager operators into the query plan to prevent updates to influence pattern matching. This example query for example, where the DELETE will influence the MATCH.

Reply:
👍